### DISCUSSION TITLE
Would like to see upload handler for TOA5 data logger files

### DISCUSSION THREAD

**Discussion by alfski** [redivis/meta#222] [Feature requests]
My facility has many thousands of time series tables that are in Campbell Scientific LoggerNet TOA5 format.

Basically this is a CSV file with a 4 line header. Example below.

It would be great to be able to upload these as Tables and have the Variables and Units/Processing pulled into the metadata table in Redivis. At the moment I've been stripping off lines 1,3,4 and uploading as CSV's, and we can keep doing this.

Line 1 provides information about the logging device itself. Always starts with "TOA5"
Line 2 is the "Variable/Column Name" header.
Line 3 the "Units" of those variables as defined in the logger. eg. "Deg C", "meters/second"
Line 4 processing the logger may have done prior to recording the value eg. 'sample', 'average', 'maximum', 'minimum'
Line 5+ the data

```
"TOA5","ROS_WS","CR3000","4909","CR3000.Std.32.03","CPU:weather_station_final_20210621.CR3","59313","Table05min"
"TIMESTAMP","RECORD","PPFD_Avg","AirTC_Avg","RH","WS_ms_Avg","WS_ms_Max","WindDir","NetSW_Avg","NetLW_Avg","NetRad_Avg","
LWmV_Avg","LWMDry_Tot","LWMCon_Tot","LWMWet_Tot"
"TS","RN","mV","Deg C","%","meters/second","m/s","degrees","W/m^2","W/m^2","W/m^2","mV","Minutes","Minutes","Minutes"
"","","Avg","Avg","Smp","Avg","Max:meters/second","Smp","Avg","Avg","Avg","Avg","Tot","Tot","Tot"
"2023-05-31 00:05:00",171545,0,6.722,100,0.5,0.71,229.5,-2.332,-16.1,-18.43,397.1,0,0,5
"2023-05-31 00:10:00",171546,0,6.522,100,0.345,0.48,261.8,-1.383,-15.62,-17,401.2,0,0,5
"2023-05-31 00:15:00",171547,0,6.432,100,0.513,0.83,202,-1.665,-14.71,-16.38,407.8,0,0,5
"2023-05-31 00:20:00",171548,0,6.186,100,0.363,0.7,169.3,-2.002,-14.54,-16.54,411.7,0,0,5
```

Cheers, Andrew (Western Sydney University)

**Comment by imathews** [comment]
Thanks @alfski. I think the "generic" solution could be to introduce new advanced options for delimited data uploads, so that you could a `data_start_row`, `variable_name_row`, `variable_label_row`, etc. Of course the fact that variable "labels" are stored across multiple rows (in this case) introduces another wrinkle...

In the interim, you could definitely script this through the [python client](https://apidocs.redivis.com/client-libraries/redivis-python/reference/variable#update-label-description-value_labels), and set corresponding variable metadata after the trimmed CSV is uploaded.
